src/main/resources/react4xp/components/PageProcessor.ts

```diff
@@ -1,17 +1,25 @@
-import type {ComponentProcessorFunction} from '@enonic-types/lib-react4xp/DataFetcher';
+import {getSite, pageUrl} from '/lib/xp/portal';
 import {PageComponent} from "@enonic-types/core";
-
+import type {ComponentProcessorFunction} from '@enonic-types/lib-react4xp/DataFetcher';
 
 export const pageProcessor: ComponentProcessorFunction<'com.enonic.app.hmdb:main'> = (props) => {
     const component = props.component as PageComponent;
     const regions = component?.regions || {};
+    const site = getSite();
+
+    const baseUrl = pageUrl({path: site._path});
+    const parentSegment = props.request.path.split('/').slice(0, -1).join('/');
```

Comment:
This line is used a lot in different processors, create a util function that accepts path and returns a parent and use it everywhere.